src/neo4j_genai/components/pdf_loader.py

```diff
@@ -0,0 +1,79 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import io
+from abc import abstractmethod
+from pathlib import Path
+from typing import Optional
+
+import fsspec
+from fsspec import AbstractFileSystem
+from fsspec.implementations.local import LocalFileSystem
+
+from neo4j_genai.exceptions import PdfLoaderError
+from neo4j_genai.pipeline import Component, DataModel
+
+
+class PdfDocument(DataModel):
+    text: str
+
+
+class DataLoader(Component):
+    """
+    Interface for loading data of various input types.
+    """
+
+    @abstractmethod
+    async def run(self, filepath: Path) -> PdfDocument:
+        pass
+
+
+def is_default_fs(fs: fsspec.AbstractFileSystem) -> bool:
+    return isinstance(fs, LocalFileSystem) and not fs.auto_mkdir
+
+
+class PdfLoader(DataLoader):
+    @staticmethod
+    def load_file(
+        file: Path,
+        fs: Optional[AbstractFileSystem] = None,
+    ) -> str:
+        """Parse PDF file and return text."""
+        if not isinstance(file, Path):
+            file = Path(file)
+
+        try:
+            import pypdf
+        except ImportError:
+            raise ImportError(
+                "pypdf is required to read PDF files: `pip install pypdf`"
+            )
+        fs = fs or LocalFileSystem()
```

Comment:
What are these `FileSystem` used for?

Reply:
It's an interface for dealing with different types of file system environments e.g. `LocalFileSystem` vs `S3FileSystem` https://filesystem-spec.readthedocs.io/en/latest/usage.html

Reply:
Great, so we already support fetching from URL!